rllib/offline/tests/test_dataset_reader.py

```diff
@@ -23,7 +23,10 @@ def setUpClass(cls) -> None:
         #  credentials issues, using a local file instead for now.
 
         # cls.dset_path = "s3://air-example-data/rllib/cartpole/large.json"
```

Comment:
@gjoliver  Does pathlib work properly with S3's syntax?

Reply:
I think we would have to add another dependency to make pathlib work here.

Reply:
<img width="523" alt="Screenshot 2022-12-01 at 01 14 59" src="https://user-images.githubusercontent.com/9356806/205013359-b147ae95-b5ab-49ff-9d67-31447c8cdb73.png">

Reply:
Tried to play with it and immediately hit this 😢

Reply:
Good catch! This is already discussed on [python-ideas](https://mail.python.org/pipermail/python-ideas/2016-March/039268.html). We could work with `urlparse`:

Reply:
So another dependency, right?

Reply:
> I like these changes. Pathlib is modern. But I think we might run into issues with cloud file paths here!

See my suggestion above using maybe `urllib`'s `urlparse`

Reply:
How would you do that? Can you include it in the PR? Because right now the PR would break usage of URIs 🤷

Reply:
I could make a PR for this. And yes, there will be another dependency towards a Python core package. Basically it is simply used to check for a non-empty scheme as if for cloud paths the path should not be resolved. This could possibly also be solved without using `urllib`